daemon/logger/copier.go

```diff
@@ -41,21 +43,20 @@ func (c *Copier) Run() {
 
 func (c *Copier) copySrc(name string, src io.Reader) {
 	defer c.copyJobs.Done()
-	reader := bufio.NewReader(src)
+	reader := bufio.NewReaderSize(src, MaxBytesPerLine)
 
 	for {
-		line, err := reader.ReadBytes('\n')
-		line = bytes.TrimSuffix(line, []byte{'\n'})
+		// ReadLine tries to return a single line, not including the end-of-line bytes.
+		// If the line was too long for the buffer then isPrefix is set and the
+		// beginning of the line is returned. The rest of the line will be returned
+		// from future calls.
+		line, _, err := reader.ReadLine()
```

Comment:
Should we not check `isPrefix` and do something about it? Silently corrupting the log message is probably not the right solution.